### PR TITLE
feat: Ensures related access is deleted after user/device deletion

### PR DESCRIPTION
**Pull request by frgfm**
This PR ensures that once a user or a device is deleted, its corresponding access is deleted as well.

I had to change the table definitions in unittests since accesses were sometimes not consistent with device/user tables.

### CODE REVIEW

File: src/app/api/routes/devices.py [hunk 1]
```diff
@@ -35,7 +35,10 @@ async def update_device(payload: DeviceIn, device_id: int = Path(..., gt=0)):
 
 @router.delete("/{device_id}/", response_model=DeviceOut)
 async def delete_device(device_id: int = Path(..., gt=0), _=Security(get_current_user, scopes=["admin"])):
```
Comment (florianriche) [inline src/app/api/routes/devices.py:37]: Can we do a "db" transaction like what JB suggested in one o his reviews?

File: src/app/api/routes/users.py [hunk 2]
```diff
@@ -71,4 +71,8 @@ async def update_user_password(
 
 @router.delete("/{user_id}/", response_model=UserRead)
 async def delete_user(user_id: int = Path(..., gt=0), _=Security(get_current_user, scopes=["admin"])):
-    return await crud.delete_entry(users, user_id)
+    # Delete user entry
+    entry = await crud.delete_entry(users, user_id)
+    # Delete access
```
Comment (florianriche) [inline src/app/api/routes/users.py:76]: same commet

File: src/app/api/routes/devices.py (hunk 1 above)
Comment (frgfm) [reply src/app/api/routes/devices.py:37]: Considering the PR is still hanging and I don't know how to use them as of now, I didn't consider this for this PR. But I still have one last thing to check with this PR (some troubles in the unittests)

File: src/tests/conftest.py [hunk 3]
```diff
@@ -15,7 +15,7 @@ async def mock_current_user():
 
 
 async def mock_current_device():
-    return DeviceOut(id=1, owner_id=1, specs="raspberry", login="connected_device", created_at=datetime.now())
+    return DeviceOut(id=3, owner_id=1, specs="raspberry", login="connected_device", created_at=datetime.now())
```
Comment (florianriche) [inline src/tests/conftest.py:18]: It might break some tests I fear if you merge with master; or maybe you already took care of it?

**Review by florianriche — APPROVED** [review]
I just have  a comment that is a question/Warning. If my comment is useless, then let's merge this PR

File: src/tests/conftest.py (hunk 3 above)
Comment (frgfm) [reply src/tests/conftest.py:18]: Yup already did, though this was cumbersome haha

**Comment by frgfm** [comment]
> I just have a comment that is a question/Warning. If my comment is useless, then let's merge this PR
> 
> **EDIT: Should we not delete user/device when deleting an access as well?**

I wondered as well, but I felt it was quite dangerous. Only admin should be able to delete accesses directly, so I thought that self deletion of user/device should trigger the deletion of their access, but the other way is quite dangerous. But if you have a suggestion, happy to check it!

**Pull request merged**